reactive/kubernetes_master.py

```diff
@@ -281,8 +283,14 @@ def post_series_upgrade():
 
 
 def add_rbac_roles():
-    '''Update the known_tokens file with proper groups.'''
+    '''Update the known_tokens file with proper groups.
 
+    DEPRECATED: Once known_tokens are migrated, group data will be stored in K8s
+    secrets. Do not use this function after migrating to authn with secrets.
+    '''
+    if is_flag_set('kubernetes-master.token-auth.migrated'):
+        hookenv.log('Known tokens have migrated to secrets. Skipping group changes')
```

Comment:
Wouldn't hitting this case imply that it's a charm code error, likely caused by a new change introducing a regression?  It feels like this should perhaps be more fatal.

Reply:
Not sure I follow.  This function ensures groups are correctly added to `known_tokens.csv`, which they still are during `upgrade-charm` for clusters < 1.19 -- in fact, `add_rbac_roles` is *only* called on upgrade. Once `k_t.csv` is migrated to k8s secrets, there will be no reason to add groups, as that information will have already been stored in the secret.

Once secrets have been around for 3 releases, `add_rbac_roles()` should be removed.

Reply:
Ah, gotcha.  I think I was confused by the last sentence of the deprecation notice.